src/services/TechRecordsService.ts

```diff
@@ -134,6 +134,7 @@ class TechRecordsService {
     if (isPayloadValid.error) {
       return Promise.reject({statusCode: 400, body: isPayloadValid.error.details});
     }
+    await this.generateSystemNumber(techRecord);
```

Comment:
as a general principle of JS, mutation of objects is not ideal. Better to return new objects with the changes

Reply:
Yes, I know we should avoid object mutation, but since we need to update the object at least 3-4 times and the changes are not that big, I thought that creating new instances may not be so beneficial and it can lead to performance issues since the structure of the object is quite large. That's the reason I preferred to mutate it. Should I return new instances from the methods where I currently mutate the object?

Reply:
Just return the system number and assign instead of object mutation.